src/chronify/store.py

```diff
@@ -28,52 +51,94 @@
 
+    SCHEMAS_TABLE = "schemas"
+
     def __init__(
         self,
         engine: Optional[Engine] = None,
         engine_name: Optional[str] = None,
+        file_path: Optional[Path | str] = None,
         **connect_kwargs: Any,
     ) -> None:
         """Construct the Store.
 
         Parameters
         ----------
-        engine: sqlalchemy.Engine
+        engine
             Optional, defaults to a engine connected to an in-memory DuckDB database.
+        engine_name
+            Optional, name of engine to use ('duckdb', 'sqlite'). Mutually exclusive with engine.
+        file_path
+            Optional, use this file for the database. If the file does not exist, create a new
+            database. If the file exists, load that existing database.
+            Defaults to a new in-memory database.
 
         Examples
         --------
         >>> from sqlalchemy
         >>> store1 = Store()
-        >>> store2 = Store(engine=Engine("duckdb:///time_series.db")
-        >>> store3 = Store(engine=Engine(engine_name="sqlite")
-        >>> store4 = Store(engine=Engine("sqlite:///time_series.db")
+        >>> store2 = Store(engine=Engine("duckdb:///time_series.db"))
+        >>> store3 = Store(engine=Engine(engine_name="sqlite"))
+        >>> store4 = Store(engine=Engine("sqlite:///time_series.db"))
         """
         self._metadata = MetaData()
         if engine and engine_name:
             msg = f"{engine=} and {engine_name=} cannot both be set"
             raise ConflictingInputsError(msg)
+        filename = ":memory:" if file_path is None else str(file_path)
         if engine is None:
             name = engine_name or "duckdb"
             match name:
                 case "duckdb" | "sqlite":
-                    engine_path = f"{name}:///:memory:"
+                    engine_path = f"{name}:///{filename}"
                 case _:
                     msg = f"{engine_name=}"
                     raise NotImplementedError(msg)
             self._engine = create_engine(engine_path, **connect_kwargs)
         else:
             self._engine = engine
 
+        self.update_table_schema()
+        if self.has_table(self.SCHEMAS_TABLE):
+            logger.info("Loaded existing database {}", file_path)
+        else:
+            table = Table(
+                self.SCHEMAS_TABLE,
+                self._metadata,
+                Column("name", String),
+                Column("schema", String),  # schema encoded as JSON
+            )
+            # TODO: will this work for Spark? Needs to be mutable.
+            table.create(self._engine)
+            self.update_table_schema()
+            logger.info("Initialized new database. file = {}", filename)
+
+    def backup(self, dst: Path | str, overwrite: bool = False) -> None:
+        """Copy the database to a new location. Not supported for in-memory databases."""
+        self._engine.dispose()
+        path = Path(dst) if isinstance(dst, str) else dst
```

Comment:
You can call Path() on a Path

Reply:
Sure, but we could still avoid it. Not sure it matters.

Reply:
Might as well keep the faster solution.
```
In [4]: path1 = Path("/dir1/dir2/dir3/dir4/dir5")
In [5]: %timeit path2 = Path(path1)
1.35 μs ± 9.69 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [6]: %timeit path2 = Path(path1) if isinstance(path1, str) else path1
28.2 ns ± 0.0859 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```